src/datasource.ts

```diff
@@ -33,27 +37,31 @@ export class CriblDataSource extends DataSourceApi<CriblQuery, CriblDataSourceOp
    */
   async query(options: DataQueryRequest<CriblQuery>): Promise<DataQueryResponse> {
     // options.targets contains one or more queries, each of which results in a DataFrame
-    return Promise.all(options.targets.map((q) => this.processQuery(q))).then((data) => ({ data }));
+    return Promise.all(options.targets.map((q) => this.processQuery(q, options.range))).then((data) => ({ data }));
   }
 
   /**
    * Run a single query to produce a single DataFrame
    * @param criblQuery the query to run
+   * @param range the time range from the time picker
    * @returns the DataFrame with the results
    */
-  private async processQuery(criblQuery: CriblQuery): Promise<DataFrame> {
+  private async processQuery(criblQuery: CriblQuery, range: TimeRange): Promise<DataFrame> {
     let fields: Record<string, Field> = {};
     let eventCount = 0;
     let totalEventCount: number | undefined = undefined;
-    let maxEventCount = criblQuery.maxResults ?? Number.MAX_VALUE;
     let startTime = Date.now();
 
-    // Initially we're simply trying to load the most recent results for the given saved search
-    let queryParams: any = {
-      queryId: criblQuery.savedSearchId,
-    };
+    let queryParams: any = criblQuery.type === 'saved'
+      ? { queryId: criblQuery.savedSearchId }
+      : {
+        query: prependCriblOperator(criblQuery.query.replace(/[\r\n\t]+/, ' ')),
+        earliest: range.from.unix(),
+        latest: range.to.unix(),
+      };
 
-    // Load the search results, paging through until we've hit maxResults or read them all, whatever comes first
+    // Load the search results, paging through until we've hit MAX_RESULTS or read them all, whatever comes first
+    const getBackoff = createGetBackoff();
     do {
       const authorization = await this.getAuthorization(); // refresh auth token as needed
```

Comment:
(tl;dr I've forgotten the exact model for Grafana plugin development)
For my understanding: is this the auth token to grafana? Or to Cribl Cloud? I would assume it's the former, since this code runs on the UI afaict, but I saw references to the Cribl Cloud instance inside getAuthorization so I wasn't sure.

Reply:
@bardhi This is the Cribl API auth token.  The code runs on UI, but all API calls are proxied through the Grafana backend.  I'll do a writeup in Confluence about this, but you'll probably remember we had to do that for 2 reasons (1) CORS madness, (2) since the plugin stores the sensitive client secret in the backend, we had to proxy thru backend for the auth calls anyway.  So it's easiest/sanest just to proxy all of it.  Even though we could get away with only proxying the auth calls, and going direct for the other calls, CORS becomes a bit of a nightmare.

Also...doing local testing is impractical, since we'd need to configure 2 different API endpoints: one the backend uses (host.docker.internal) and another that the UI uses (localhost).

We just proxy everything.